Update README.md (#9)
actions-on-packages/.github: profile/README.md

With Actions Packages, you can reference actions by their version and feel confident that you are using the exact version as specified when your workflow runs.

> ## How to get started with `Actions on packages`

Reviewer: Do we need to mention about updating the visibility settings at Org level?

Author: I believe there will be an appropriate error reported if org settings are not as expected right ?

> - To create Actions on packages, you need to copy the [release.yml](https://github.com/actions-on-packages/.github/blob/main/workflow-templates/release.yml) workflow from this repository. This workflow is responsible for packaging the action contents  📦 while creating new releases and publish them to a container registry 🛳️.

Reviewer 2: ```suggestion
- To use Actions on packages, you need to copy the [release.yml](https://github.com/actions-on-packages/.github/blob/main/workflow-templates/release.yml) workflow from this repository. This workflow is responsible for packaging the action contents  📦 while creating new releases and publish them to a container registry 🛳️.
```

> - To create Actions on packages, you need to copy the [release.yml](https://github.com/actions-on-packages/.github/blob/main/workflow-templates/release.yml) workflow from this repository. This workflow is responsible for packaging the action contents  📦 while creating new releases and publish them to a container registry 🛳️.

Reviewer 2: ```suggestion
- To create an action package, you need to copy the [release.yml](https://github.com/actions-on-packages/.github/blob/main/workflow-templates/release.yml) workflow from this repository. This workflow is responsible for packaging the action contents  📦 while creating new releases and publish them to a container registry 🛳️.
```

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/13884596/207019455-43b2f166-21b8-41ca-8ebf-f4d64546e668.png">

## How to use `actions-packages` in workflows
The repository which contains the consuming workflow referencing some actions needs to be added to FF `actions_on_packages`, which also allows action traffic to be routed to the actions-on-packages. Please leave a comment in the [discussion here](https://github.com/github/c2c-actions-nirvana/discussions/267) if you would like to try out Actions on packages or if you have any suggestions/feedback.

### Migrated releases
The migrated releases will continue to use the same referencing semantics, so there is no need to update the existing workflows.

Say: If your consuming workflow is referencing
 ```shell
    uses: mycool\myawesome@v1.0.0
 ``` 

> and if the action creator of mycool\myawesome has migrated the v1.0.0 release to actions on packages, there is no need to update the existing consuming workflow. Your consuming workflow repo once added to the FF `actions_on_packages will automatically resolve actions via actions on packages. Also refer [splat versioning](https://github.com/actions-on-packages/.github#splat-versioning) if your consuming workflow has reference to actions which are pinned to major versions like **uses: mycool\myawesome@v4**

Reviewer 2: imo this information is overwhelming. I think what we had earlier is concise.

Author: I felt it is not information hence added more detail to avoid further questions for a new comer

Reviewer 2: pls remove this if above suggested commit is accepted.

Author: I would like to have that detail. Hence i will not be able to remove the additional detail

Author: This is anyway no official doc. This is for github folks only

Reviewer 2: okay. You might want to clean up ###Migrated releases section as we've duplicate info now.

Author: Updated. When i re-read i too felt some text is overwhelming and did pertaining changes

### New releases

> The new release or migrated release generates the `action-package` which can be found under the associated packages on the repository home page. Navigate to the action-package and you will find the usage snippet. See the screenshots below:

Reviewer 2: ```suggestion
Once a new or migrated release is generated, an associated action-package will be created and can be found on the repository's home page under the associated packages section. You can navigate to the action-package and find the usage snippet there. Take a look at the screenshots below for reference:
```

Reviewer 2: I'm not sure if we want to name it `action-package` or `action package` or `package action` 😕 😅

<img width="768" alt="image" src="https://user-images.githubusercontent.com/17195847/207063008-141b9c4d-795b-4a9d-860e-8c0fd76c74c7.png">
<img width="767" alt="image" src="https://user-images.githubusercontent.com/17195847/207068051-576ade19-78fe-4555-99bd-48215128e765.png">

### Consuming Private and Internal Actions Packages

> To consume Private and Internal Action Packages, the consuming workflow repo needs to be provided access via Package settings. This is very important to do this step before you attempt consumption of Private and Internal Action Packages. If your workflow is consuming Public Action Packages no action required.

Reviewer 2: ```suggestion
To use private and internal action packages, you must first grant access to the consuming workflow repository via Package settings. It's important to perform this step before attempting to use them. If your workflow uses public action packages, then no further action is necessary.
```

> #### Private Action Package Access 

Reviewer 2: ```suggestion
<details>
  <summary>To access a private action package</summary>
  
    <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815402-ba5ba148-1043-4a4d-a814-03de5748b446.png">
    <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815469-d112e5c8-4d43-461d-991d-7762eca98d0a.png">
</details>
```

![private-action-package-settings-black](https://user-images.githubusercontent.com/100673363/225815402-ba5ba148-1043-4a4d-a814-03de5748b446.png)
![access-to-private-action-black](https://user-images.githubusercontent.com/100673363/225815469-d112e5c8-4d43-461d-991d-7762eca98d0a.png)
#### Internal Action Package Access 
![internal-action-package-settings-black](https://user-images.githubusercontent.com/100673363/225815622-fa7a35fe-14a0-4297-829b-481768e67e8f.png)
![access-to-internal-action-black](https://user-images.githubusercontent.com/100673363/225815637-d94c5090-8151-41fb-8351-141d3e23fd9c.png)

### Splat versioning
🚀Good news🚀 Actions-Packages also supports the splat versioning hence the actions can be referenced using the sample semantics without having any additional tag maintained.

> Say: The action author of `mycool\myawesome` just released a new version action version `v4.2.2` and an action package mycool\myawesome:4.2.2 is also created as part of [release process](https://github.com/actions-on-packages/.github#how-to-get-started-with-actions-on-packages).

Reviewer 2: ```suggestion
Say: A new version v4.2.2 of the mycool\myawesome action has been released along with an associated action package mycool\myawesome:4.2.2 created during the [release process](https://github.com/actions-on-packages/.github#how-to-get-started-with-actions-on-packages).
```

## How to use `actions-packages` in workflows
The repository which contains the consuming workflow referencing some actions needs to be added to FF `actions_on_packages`, which also allows action traffic to be routed to the actions-on-packages. Please leave a comment in the [discussion here](https://github.com/github/c2c-actions-nirvana/discussions/267) if you would like to try out Actions on packages or if you have any suggestions/feedback.

### Migrated releases
If a consuming workflow references an action release say `uses: mycool\myawesome@v1.0.0` and that release is migrated to Actions on Packages, there is no need to update the existing workflow. The migrated releases will continue to use the same referencing semantics. Once your consuming workflow repo is added to the FF `actions_on_packages`, it will automatically resolve actions via Actions on Packages. 

Also, if your consuming workflow references actions pinned to major versions, such as `uses: mycool\myawesome@v4`, refer to [splat versioning](https://github.com/actions-on-packages/.github#splat-versioning).

### New releases

> Once a new or migrated release is generated, an associated action package will be created and can be found on the repository's home page under the associated packages section. You can navigate to the action package and find the usage snippet there. Take a look at the screenshots below for reference:

Reviewer 2: ```suggestion
Once a new or migrated release is generated, an associated action package will be created and can be found on the repository's home page under the associated packages section. You can navigate to the action package and find the usage snippet there. Take a look at the screenshots below for reference:

```

Reviewer 2: fixes image alignment.

### Consuming Private and Internal Actions Packages
To use private and internal action packages, you must first grant access to the consuming workflow repository via Package settings. It's important to perform this step before attempting to use them. If your workflow uses public action packages, then no further action is necessary.

<details>
  <summary>To access a private action package</summary>
  
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815402-ba5ba148-1043-4a4d-a814-03de5748b446.png">
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815469-d112e5c8-4d43-461d-991d-7762eca98d0a.png">
</details>
&nbsp;

<details>
  <summary>To access an internal action package</summary>
  
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815622-fa7a35fe-14a0-4297-829b-481768e67e8f.png">
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815637-d94c5090-8151-41fb-8351-141d3e23fd9c.png">
</details>

> &nbsp;

Reviewer 2: ```suggestion
```

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/13884596/207019455-43b2f166-21b8-41ca-8ebf-f4d64546e668.png">

## How to use `actions-packages` in workflows
The repository which contains the consuming workflow referencing some actions needs to be added to FF `actions_on_packages`, which also allows action traffic to be routed to the actions-on-packages. Please leave a comment in the [discussion here](https://github.com/github/c2c-actions-nirvana/discussions/267) if you would like to try out Actions on packages or if you have any suggestions/feedback.

### Migrated releases

> The migrated releases will continue to use the same referencing semantics, so there is no need to update the existing workflows.

Reviewer 2: nit: we can simplify it a bit, if decision is to include above details
```suggestion
If a consuming workflow references an action release and that release is migrated to Actions on Packages, there is no need to update the existing workflow. The migrated releases will continue to use the same referencing semantics. Once your consuming workflow repo is added to the FF `actions_on_packages`, it will automatically resolve actions via Actions on Packages. Also, If your consuming workflow references actions pinned to major versions, such as `uses: mycool\myawesome@v4`, refer to [splat versioning](https://github.com/actions-on-packages/.github#splat-versioning).
```

### Consuming Private and Internal Actions Packages
To use private and internal action packages, you must first grant access to the consuming workflow repository via Package settings. It's important to perform this step before attempting to use them. If your workflow uses public action packages, then no further action is necessary.

<details>
  <summary>To access a private action package</summary>
  
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815402-ba5ba148-1043-4a4d-a814-03de5748b446.png">
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815469-d112e5c8-4d43-461d-991d-7762eca98d0a.png">
</details>
&nbsp;

<details>
  <summary>To access an internal action package</summary>
  
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815622-fa7a35fe-14a0-4297-829b-481768e67e8f.png">
  <img width="768" alt="image" src="https://user-images.githubusercontent.com/100673363/225815637-d94c5090-8151-41fb-8351-141d3e23fd9c.png">
</details>
&nbsp;

> 

Reviewer 2: ```suggestion
```
